pkgs/development/tools/k6/xk6.nix

```diff
@@ -0,0 +1,32 @@
+{ lib, buildGoModule, fetchFromGitHub, makeWrapper, go }:
+
+buildGoModule rec {
+  pname = "xk6";
+  version = "0.6.1";
+
+  src = fetchFromGitHub {
+    owner = "grafana";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-3jA4e7EL/7wMETjX15JMmTMLCb+b+4MW6XhHeyj+dO8=";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  subPackages = [ "./cmd/xk6" ];
+
+  vendorSha256 = null;
+
+  doCheck = false;
+
+  postInstall = ''
+    wrapProgram $out/bin/xk6 --prefix PATH : ${lib.makeBinPath [ go ]}
```

Comment:
while I support this change, I got called out in #202592 that "maybe a user wants to bring their own custom version of go"; feel free to leave this, as `xk6` may be non-functional without `go`, but fyi